### PR TITLE
[4.0] Add the username to the statusbar

### PR DESCRIPTION
**Pull request by chmst**
### Summary of Changes
Add the username in the backend status bar. 

user-name

### CODE REVIEW

**Comment by richard67** [comment]
To be honest, for me it is ok like it is now, I see "User Menu", and when I click on it I see my name. We will see what others say.

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on 808031f882c950b04692ba28ff93f681d7e9f6cc<br /><br />it's OK for me - both :-)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26481">issues.joomla.org/tracker/joomla-cms/26481</a>.</sub>

**Comment by chmst** [comment]
I use to play different roles when working on a site to be sure that all ACL is correct. Somteimes as super user, sometimes as a manger or in another role. And use to forget which role I am playing. :)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26481">issues.joomla.org/tracker/joomla-cms/26481</a>.</sub>

**Comment by ChristineWk** [comment]
Of course, this makes sense :-)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26481">issues.joomla.org/tracker/joomla-cms/26481</a>.</sub>

File: administrator/modules/mod_user/tmpl/default.php [hunk 1]
```diff
@@ -24,7 +25,7 @@
 			<span class="fa fa-user-circle" aria-hidden="true"></span>
 		</div>
 		<div class="tiny">
-			<?php echo Text::_('MOD_USER_MENU'); ?>
+			<?php echo htmlspecialchars(Factory::getUser()->name); ?>
```
Comment (SharkyKZ) [inline administrator/modules/mod_user/tmpl/default.php]:
```suggestion
			<?php echo htmlspecialchars($app->getIdentity()->name); ?>
```
And remove `use Joomla\CMS\Factory;`.

Comment (chmst) [reply administrator/modules/mod_user/tmpl/default.php]: ty

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on 72d87fa804cf214f102f2f2fe6724a33eb150da6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26481">issues.joomla.org/tracker/joomla-cms/26481</a>.</sub>

**Comment by brianteeman** [comment]
There is an a11y issue with this chambe as the drop-down menu no longer has a sensible description

**Comment by chmst** [comment]
@brianteeman you are right. It is a pitty, it would be more friendly and individual, if the users sees his/her name. 
I can make a Text like this "Pierre Dupont, your User Menu" but this will be truncated in the module.

**Comment by astridx** [comment]
@chmst Perhaps you can use 'sr-only' for the long text?

**Comment by brianteeman** [comment]
I really think your use case is an exception and not a norm

**Comment by richard67** [comment]
By the way: In database, the user name column can be up to 400 characters long. What if someone has a very long name? Shall that really all be displayed at that place?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26481">issues.joomla.org/tracker/joomla-cms/26481</a>.</sub>

**Comment by alikon** [comment]
I like this concept, despite  i cannot suggest a more good way to do this, but i really would like to know in a easy way --- `who am I`

**Comment by richard67** [comment]
Who am I ... and how many? ;-)

**Comment by brianteeman** [comment]
@alikon not really that hard :)
image

**Comment by alikon** [comment]
philosophical question apart, :smile: 
 make sense to me... but
i know i'm not so good at UI/UX  just my 2eurocent

**Comment by chmst** [comment]
@richard67 Long names would be truncated .

@brianteeman  I can make an override of the module for myself, and I do not insist to bring this into the core.
But I see in different applications my avatar and/or user name, and I like it, it is more personal and friendly.

I'll commit a small change and with my long name it looks like this:

username

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on 5555b2bf7b5e9971de1ec97457d15bc8a4872d05<br /><br />OK, I gave it a 3rd try - but have short User Name :-)
So, it looks OK, even it's truncated - in your case. What about user, inserting as User Name: Super User :-)

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26481">issues.joomla.org/tracker/joomla-cms/26481</a>.</sub>

**Comment by Chessy65** [comment]
I have tested this item white_check_mark successfully on 5555b2bf7b

File: administrator/language/en-GB/en-GB.mod_user.ini [hunk 2]
```diff
@@ -6,5 +6,5 @@
 MOD_USER="User Menu"
 MOD_USER_ACCESSIBILITY_SETTINGS="Accessibility Settings"
 MOD_USER_EDIT_ACCOUNT="Edit Account"
-MOD_USER_MENU="User Menu"
+MOD_USER_MENU="User Menu of  %s"
```
Comment (Quy) [inline administrator/language/en-GB/en-GB.mod_user.ini]:
```suggestion
MOD_USER_MENU="User Menu of %s"
```

**Comment by Quy** [comment]
I find this wordy. I like the way before this PR.

**Comment by scott-qld** [comment]
I have tested this item :red_circle: unsuccessfully on 82576dc0eb305128bb7de3d98ccddbaad9d4040a<br /><br />Patch could not be applied due to exception with administrator/components/com_media/package-lock.json. Joomla\Filesystem\File::move: Rename failed.
PHP Built On 	Linux joomlapbf.directrouter.com 2.6.32-754.3.5.el6.x86_64 #1 SMP Tue Aug 14 20:46:41 UTC 2018 x86_64
Database Type 	mysql
Database Version 	10.2.27-MariaDB
Database Collation 	utf8mb4_unicode_ci
Database Connection Collation 	utf8mb4_general_ci
PHP Version 	7.2.23
Web Server 	Apache
WebServer to PHP Interface 	cgi-fcgi
Joomla! Version 	Joomla! 4.0.0-beta1-dev Development [ Amani ] 17-October-2019 20:21 GMT
User Agent 	Mozilla/5.0 (X11; Linux x86_64; rv:68.0) Gecko/20100101 Firefox/68.0 <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26481">issues.joomla.org/tracker/joomla-cms/26481</a>.</sub>

**Comment by manojLondhe** [comment]
I have tested this item :white_check_mark: successfully on 82576dc0eb305128bb7de3d98ccddbaad9d4040a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26481">issues.joomla.org/tracker/joomla-cms/26481</a>.</sub>

**Comment by infograf768** [comment]
Agree with @Quy 
The User Name is immediately visible in the dropdown when clicking on the icon. No need to change imho.

**Comment by uthorat** [comment]
I have tested this item :white_check_mark: successfully on 82576dc0eb305128bb7de3d98ccddbaad9d4040a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26481">issues.joomla.org/tracker/joomla-cms/26481</a>.</sub>

**Comment by idefax** [comment]
I have tested this item :white_check_mark: successfully on 82576dc0eb305128bb7de3d98ccddbaad9d4040a<br /><br />Works fine.

Tested on:

PHP Version 	7.3.8
Web Server 	Apache
WebServer to PHP Interface 	cgi-fcgi
Joomla! Version 	Joomla! 4.0.0-beta1-dev Development [ Amani ] 17-October-2019 20:21 GMT
User Agent 	Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:69.0) Gecko/20100101 Firefox/69.0 <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26481">issues.joomla.org/tracker/joomla-cms/26481</a>.</sub>

**Comment by brianteeman** [comment]
I am not in favour of this change

**Comment by richard67** [comment]
Me neither, but I won't block it if many people want it.

**Comment by snehaM26** [comment]
I have tested this item :white_check_mark: successfully on 82576dc0eb305128bb7de3d98ccddbaad9d4040a<br /><br />Tested successfully<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26481">issues.joomla.org/tracker/joomla-cms/26481</a>.</sub>

**Comment by LukasHH** [comment]
I have tested this item :white_check_mark: successfully on 82576dc0eb305128bb7de3d98ccddbaad9d4040a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26481">issues.joomla.org/tracker/joomla-cms/26481</a>.</sub>

**Comment by neotrendch** [comment]
I have tested this item :white_check_mark: successfully on 82576dc0eb305128bb7de3d98ccddbaad9d4040a<br /><br />good to see the username without click<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26481">issues.joomla.org/tracker/joomla-cms/26481</a>.</sub>

**Comment by richard67** [comment]
Guys, 2 successful tests are enough for a Pull Request to get the "Ready to Commit" status. As soon as a PR has this, it does not need more tests, and you can focus on other PR's which still need tests.

**Comment by oschuldt** [comment]
I have tested this item :white_check_mark: successfully on 82576dc0eb305128bb7de3d98ccddbaad9d4040a<br /><br />works!

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26481">issues.joomla.org/tracker/joomla-cms/26481</a>.</sub>

**Comment by patjun** [comment]
I have tested this item :white_check_mark: successfully on 82576dc0eb305128bb7de3d98ccddbaad9d4040a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26481">issues.joomla.org/tracker/joomla-cms/26481</a>.</sub>

**Comment by karo3** [comment]
I have tested this item :white_check_mark: successfully on 82576dc0eb305128bb7de3d98ccddbaad9d4040a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26481">issues.joomla.org/tracker/joomla-cms/26481</a>.</sub>

**Comment by richard67** [comment]
Guys, 2 successful tests are enough for a Pull Request to get the "Ready to Commit" status. As soon as a PR has this, it does not need more tests, and you can focus on other PR's which still need tests.

**Comment by riishede** [comment]
I have tested this item succesfully on 82576dc0eb<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26481">issues.joomla.org/tracker/joomla-cms/26481</a>.</sub>

**Comment by CodeOfConduct** [comment]
I have tested this item :white_check_mark: successfully on 82576dc0eb305128bb7de3d98ccddbaad9d4040a<br /><br />It worked for me :).<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26481">issues.joomla.org/tracker/joomla-cms/26481</a>.</sub>

**Comment by pascalelocher** [comment]
I have tested this item :white_check_mark: successfully on 82576dc0eb305128bb7de3d98ccddbaad9d4040a<br /><br />PHP Version 	7.3.8
Web Server 	Apache
WebServer to PHP Interface 	cgi-fcgi
Joomla! Version 	Joomla! 4.0.0-beta1-dev Development [ Amani ] 17-October-2019 20:21 GMT
User Agent 	Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:69.0) Gecko/20100101 Firefox/69.0 <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26481">issues.joomla.org/tracker/joomla-cms/26481</a>.</sub>

**Comment by blueforce** [comment]
I have tested this item :white_check_mark: successfully on 82576dc0eb305128bb7de3d98ccddbaad9d4040a<br /><br />works as expected
tested with:
PHP Version	7.3.8
Web Server	Apache
WebServer to PHP Interface	cgi-fcgi
Joomla! Version	Joomla! 4.0.0-alpha12-dev Development [ Amani ] 19-August-2019 13:40 GMT
User Agent	Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.120 Safari/537.36
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26481">issues.joomla.org/tracker/joomla-cms/26481</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26481">issues.joomla.org/tracker/joomla-cms/26481</a>.</sub>

**Comment by Quy** [comment]
There is 1 suggested change.

**Comment by chmst** [comment]
I think it is a11y now, without further changes? @brianteeman 
@astridx - A sr-only text was a good idea, but now is not necessary, as the name and function both are visible on the dropdown.

But I do not insist on this PR, I can make overrides :)  

I would even like to have an avatar here like github to make the backend individual, but it is not possible in this version.

**Comment by brianteeman** [comment]
1. There is a change requested by @quy that is valid and must be addressed before this can be considered for merging

2. As someone with a long name themself I am not a fan of seeing my name truncated and nor do I see any benefit

3. As I was asked - I do not see any accessibility issue with this pr

**Comment by chmst** [comment]
As this is a matter of taste and everyone can make an override for its own, I close this for now.

**Pull request closed without merging**